internal/route/always_route_filtering_strategy.go

```diff
@@ -0,0 +1,133 @@
+package route
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/satsuma-data/node-gateway/internal/config"
+	"github.com/satsuma-data/node-gateway/internal/metadata"
+	"github.com/satsuma-data/node-gateway/internal/types"
+	"go.uber.org/zap"
+)
+
+type AlwaysRouteFilteringStrategy struct {
+	BackingStrategy  RoutingStrategy
+	Logger           *zap.Logger
+	NodeFilters      []NodeFilter
+	RemovableFilters []NodeFilterType
+}
+
+func (s *AlwaysRouteFilteringStrategy) RouteNextRequest(
+	upstreamsByPriority types.PriorityToUpstreamsMap,
+	requestMetadata metadata.RequestMetadata,
+) (string, error) {
+	// Create a copy of removable filters to avoid modifying the original slice.
+	removableFilters := make([]NodeFilterType, len(s.RemovableFilters))
+	copy(removableFilters, s.RemovableFilters)
+
+	filters := s.NodeFilters // WARNING: Slice assignment, not a copy!
```

Comment:
Shouldn't we make a copy of `filters`? Otherwise we're modifying the slice `s.NodeFilters`, which could affect other concurrent requests

Reply:
Oh nice, thanks for the clarification!